extension/observer/kafkatopicsobserver/extension.go

```diff
@@ -15,17 +15,17 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/endpointswatcher"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/kafka"
 )
 
 var (
-	_ extension.Extension      = (*kafkaTopicsObserver)(nil)
-	_ observer.EndpointsLister = (*kafkaTopicsObserver)(nil)
```

Comment:
There are a couple of these compile time assertions removed, is there a reason they were removed?

Reply:
I removed the `observer.EndpointsLister` assertions because that's the contract between the implementations and EndpointsWatcher -- it's an internal detail of the observer. It's only the `observer.Observable` interface that needs to be implemented for consumers (namely `receivercreator`: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/2337fc5d01930d820671790431eeaca5fa0d5229/receiver/receivercreator/receiver.go#L70)